### DISCUSSION TITLE
자바스크립트 메모리

### DISCUSSION THREAD

**Discussion by ShinYoung-Kim** [MDR-eact/modern-react-deep-dive#11] [General]
### 서론
사실 진짜 알고 싶었던 것은 JS에서 메모리 저장이 일어나는 전체 과정이었는데, 아마 엔진마다 다르게 동작하는지 해당 과정에 대한 자세한 정보는 얻을 수 없었음.
+) 관련하여 인터넷에 검색하면 JS 메모리 중 스택/힙 영역에는 각각 어떤 데이터가 저장되고, 메모리 누수를 방지하기 위한 방법이나 Garbage collector가 어떤 알고리즘으로 동작하고 있는지에 대한 내용이 주로 나옴
+) 최근 코어 JS를 동시에 독서하고 있었는데, 해당 책은 JS의 변수/데이터 영역을 나누어 데이터의 저장을 설명하고 있었고, 좀 더 단순한 GC에 대해 설명하고 있었음
-> 코어 JS를 기반으로 이해한 내용을 정리를 하였고, JS 메모리에 대해 변수/데이터 영역에 대한 정보와 스택/힙 영역에 대한 정보를 합쳐서 정리해봄.

### 내용
(이미지가 필요해서 삼성 노트에 정리함!)
![image](https://github.com/MDR-eact/modern-react-deep-dive/assets/83866983/0b745ff6-81b2-4139-95bc-5a374d48ea28)

### 출처
- (대부분) 코어 JS 책
- [아티클1](https://velog.io/@sejinkim/%EC%9E%90%EB%B0%94%EC%8A%A4%ED%81%AC%EB%A6%BD%ED%8A%B8%EC%9D%98-%EB%A9%94%EB%AA%A8%EB%A6%AC-%EA%B4%80%EB%A6%AC-%EC%84%A4%EB%AA%85)
- [아티클2](https://velog.io/@imnotmoon/JavaScript-V8-Engine%EC%9D%B4-%EB%A9%94%EB%AA%A8%EB%A6%AC%EB%A5%BC-%EA%B4%80%EB%A6%AC%ED%95%98%EB%8A%94-%EB%B0%A9%EB%B2%95)

**Comment by ShinYoung-Kim** [comment]
다른 자료 찾으시는 분들 공유해주시면 너무 고맙겠습니다

**Comment by gs0428** [comment]
트리 쉐이킹에 쓰이는 알고리즘이랑 가비지 컬렉터랑 알고리즘이 비슷할지 의문이네요!

**Comment by coddingyun** [comment]
클로져가 실행 컨텍스트 개념에서 확장되는 느낌인데 자료를 찾아보면, 정의를 명확하게 해둔데가 없더라구용.. 클로져의 의미와 구체적인 사용할 수 있는 사례가 있을 까요?
